src/lsm/lsm_work_unit.c

```diff
@@ -117,17 +117,13 @@ __wt_lsm_get_chunk_to_flush(WT_SESSION_IMPL *session,
 	} else
 		chunk = (evict_chunk != NULL) ? evict_chunk : flush_chunk;
 
-	if (chunk != NULL) {
```

Comment:
I see it. The ```chunk != NULL``` test can't fire because ret is only ever set if the chunk is non-null.

I don't really feel like fixing that one - it's an unnecessary check now, but if that code changes in the future it'd be a hard bug to find.